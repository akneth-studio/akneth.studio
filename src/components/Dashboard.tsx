import { createClient } from '@supabase/supabase-js'
import Link from 'next/link';

const supabase = createClient(
    process.env.NEXT_PUBLIC_SUPABASE_URL!,
    process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!
)

export default async function Dashboard() {
    const { data: messages, error } = await supabase
        .from('messages')
        .select('id, name, company, email, subject, message, created_at, replied, reply_text')
        .order('created_at', { ascending: false })
        .limit(5)

    if (error) {
        return <div className="alert alert-danger">Błąd ładowania wiadomości: {error.message}</div>
    }

    return (
        <section className='messg'>
            <h2 className="mb-4">Wiadomości z formularza kontaktowego</h2>
            {(!messages || messages.length === 0) ? (
                <div className="alert alert-info">Brak wiadomości.</div>
            ) : (
                <div className="table-responsive">
                    <table className="table table-striped align-middle">
                        <thead>
                            <tr>
                                <th>Data</th>
                                <th>Imię i nazwisko</th>
                                <th>Firma</th>
                                <th>E-mail</th>
                                <th>Temat</th>
                                <th>Treść</th>
                                <th>Odpowiedziano?</th>
                                <th>Treść odpowiedzi</th>
                            </tr>
                        </thead>
                        <tbody>
                            {messages.map(msg => (
                                <tr key={msg.id}>
                                    <td style={{ whiteSpace: 'nowrap' }}>
                                        {msg.created_at
                                            ? new Date(msg.created_at).toLocaleString('pl-PL', {
                                                day: '2-digit',
                                                month: '2-digit',
                                                year: 'numeric',
                                                hour: '2-digit',
                                                minute: '2-digit'
                                            })
                                            : '—'}
                                    </td>
                                    <td>{msg.name || <span className="text-muted">—</span>}</td>
                                    <td>{msg.company || <span className="text-muted">—</span>}</td>
                                    <td>
                                        {msg.email ? (
                                            <a href={`mailto:${msg.email}`}>{msg.email}</a>
                                        ) : (
                                            <span className="text-muted">—</span>
                                        )}
                                    </td>
                                    <td>{msg.subject || <span className="text-muted">—</span>}</td>
                                    <td style={{ maxWidth: 320, whiteSpace: 'pre-line', wordBreak: 'break-word' }}>
                                        {msg.message || <span className="text-muted">—</span>}
                                    </td>
                                    <td>
                                        {msg.replied
                                            ? <span className="badge bg-success">Tak</span>
                                            : <span className="badge bg-secondary">Nie</span>
                                        }
                                    </td>
                                    <td style={{ maxWidth: 320, whiteSpace: 'pre-line', wordBreak: 'break-word' }}>
                                        {msg.reply_text || <span className="text-muted">—</span>}
                                    </td>
                                </tr>
                            ))}
                        </tbody>
                    </table>
                </div>
            )}
            <div className="text-end mt-3">
                <Link href="/admin/messages" className="btn btn-outline-primary">
                    Więcej &rarr;
                </Link>
            </div>
        </section>
    )
}
