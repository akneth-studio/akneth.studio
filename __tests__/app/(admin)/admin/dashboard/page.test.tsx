/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import Dashboard from '@/app/(admin)/admin/dashboard/page';

// Mock the Auth component
jest.mock('@/components/admin/auth', () => {
  return {
    __esModule: true,
    default: jest.fn(({ children }) => {
      return React.createElement('div', { 'data-testid': 'auth-wrapper' }, children);
    })
  };
});

// Mock the MessagesPreview component
jest.mock('@/components/admin/MessagesPreview', () => {
  return {
    __esModule: true,
    default: jest.fn(() => {
      return React.createElement('div', { 'data-testid': 'messages-preview' }, 'Messages Preview Component');
    })
  };
});

// Mock the Summary component
jest.mock('@/components/admin/Summary', () => {
  return {
    __esModule: true,
    default: jest.fn(() => {
      return React.createElement('div', { 'data-testid': 'summary' }, 'Summary Component');
    })
  };
});

describe('Dashboard Page', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders the dashboard page with all components', async () => {
    const DashboardComponent = await Dashboard();
    render(DashboardComponent);

    // Check if the main heading is rendered
    expect(screen.getByText('DASHBOARD')).toBeInTheDocument();

    // Check if Auth wrapper is rendered
    expect(screen.getByTestId('auth-wrapper')).toBeInTheDocument();

    // Check if Summary component is rendered
    expect(screen.getByTestId('summary')).toBeInTheDocument();

    // Check if MessagesPreview component is rendered
    expect(screen.getByTestId('messages-preview')).toBeInTheDocument();
  });

  it('renders components in the correct order', async () => {
    const DashboardComponent = await Dashboard();
    render(DashboardComponent);

    const elements = screen.getByTestId('auth-wrapper').children;
    
    // Check that h1 comes first
    expect(elements[0].tagName).toBe('H1');
    expect(elements[0]).toHaveTextContent('DASHBOARD');
    
    // Check that Summary comes second
    expect(elements[1]).toHaveTextContent('Summary Component');
    
    // Check that MessagesPreview comes third
    expect(elements[2]).toHaveTextContent('Messages Preview Component');
  });
  
  it('matches snapshot', async () => {
    const DashboardComponent = await Dashboard();
    const { container } = render(DashboardComponent);
    expect(container).toMatchSnapshot();
  });
});
