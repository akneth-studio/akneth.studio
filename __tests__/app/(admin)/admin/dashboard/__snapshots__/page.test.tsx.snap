// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Dashboard Page matches snapshot 1`] = `
<div>
  <div
    data-testid="auth-wrapper"
  >
    <h1
      class="text-center"
    >
      DASHBOARD
    </h1>
    <div
      data-testid="summary"
    >
      Summary Component
    </div>
    <div
      data-testid="messages-preview"
    >
      Messages Preview Component
    </div>
  </div>
</div>
`;
